mix.exs

```diff
@@ -4,33 +4,46 @@ defmodule UAParser.Mixfile do
   @version "1.0.0"
 
   def project do
-    [app: :ua_parser,
-     description: "Parse user-agent strings with BrowserScope patterns",
-     version: @version,
-     elixir: "~> 1.3",
-     build_embedded: Mix.env == :prod,
-     start_permanent: Mix.env == :prod,
-     package: package(),
-     deps: deps()]
+    [
+      app: :ua_parser,
+      description: "Parse user-agent strings with BrowserScope patterns",
+      version: @version,
+      elixir: "~> 1.3",
+      build_embedded: Mix.env == :prod,
+      start_permanent: Mix.env == :prod,
+      package: package(),
+      deps: deps()
+    ]
   end
 
   def application do
-    [applications: [:logger, :yamerl],
-     mod: {UAParser, []}]
+    [
+      applications: [:logger, :yamerl],
+      env: env(),
+      mod: {UAParser, []},
+    ]
   end
 
   defp deps do
     [
       {:yamerl, "~> 0.4.0"},
       {:credo, "~> 0.5", only: [:dev, :test]},
-      {:ex_doc, ">= 0.0.0", only: :dev}
+      {:ex_doc, ">= 0.0.0", only: :dev},
     ]
   end
 
   defp package do
-    [maintainers: ["Sean Callan", "Bobby Grayson"],
-     files: ["lib", "mix.exs", "README*", "LICENSE*", "patterns.yml"],
-     licenses: ["Apache 2.0"],
-     links: %{github: "https://github.com/doomspork/ua_parser"}]
+    [
+      maintainers: ["Sean Callan", "Bobby Grayson"],
+      files: ["lib", "mix.exs", "README*", "LICENSE*", "patterns.yml", "priv"],
+      licenses: ["Apache 2.0"],
+      links: %{github: "https://github.com/doomspork/ua_parser"}
+    ]
+  end
+
+  defp env do
+    [
+      patterns: "./priv/patterns.yml",
```

Comment:
@ybur-yug this may not work exactly as written because it assumes the locality of the `priv` directory in relationship to the configuration directive. What I recommend you do instead is have the runtime code fetch the configuration parameter and use the `priv` directory as a default.

```elixir
@spec get_patterns_filename() :: String.t
def get_patterns_filename() do
  priv_path =
    :ua_parser
    |> :code.priv_dir()
    |> IO.chardata_to_string()

  default_path = priv_path <> "/patterns.yml"
  Application.get_env(:ua_parser, :patterns, default_path)
end
```

You could of course reform the above as a case statement if you so choose.